pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -684,38 +686,25 @@ pub mod pallet {
 			let sum_of_issuances = pool_details
 				.bonded_currencies
 				.into_iter()
-				.fold(FungiblesBalanceOf::<T>::zero(), |sum, id| {
-					sum.saturating_add(T::Fungibles::total_issuance(id))
+				.fold(U256::from(0), |sum, id| {
+					sum.saturating_add(T::Fungibles::total_issuance(id).into())
 				});
 
 			defensive_assert!(
 				sum_of_issuances >= burnt,
 				"burnt amount exceeds the total supply of all bonded currencies"
 			);
```

Comment:
This can panic or not?

Reply:
this does not panic in production - it does though in non-optimised builds